client/src/research/InteractionsView.js

```diff
@@ -45,27 +76,51 @@ class InteractionsView extends Component {
     return sum/interactions.length*100;
   }
 
-  renderPercentRightPerProfile(interactions){
-    var totals = {};
-    var percents = {};
+  renderPercentRightPerProfile(interactions, key){
+    const percents = this.percentRightPerProfile(interactions, key);
+    return <pre> {JSON.stringify(percents, null,2)} </pre>;
+  }
+  renderBarChart(interactions, key, title){
+    const p = this.percentRightPerProfile(interactions, key);
+    const swipes = this.totalSwipes(interactions, key); 
+    var keys=[]; 
+    var d =[]; 
+    var count = 0;
+    var values = [];
     interactions.forEach(row =>{
-      if (row.interaction.type === InteractionTypes.SWIPE_RIGHT || row.interaction.type === InteractionTypes.SWIPE_LEFT){
-        if (row.interaction.turn.profileKey in totals){
-          if (row.interaction.type === InteractionTypes.SWIPE_RIGHT) totals[row.interaction.turn.profileKey][0] += 1;
-          totals[row.interaction.turn.profileKey][1] += 1;
-        }
-        else{
-          totals[row.interaction.turn.profileKey] = [1,1];
-        }
-      }
-      for (var k in totals){
-        percents[k] = totals[k][0]/totals[k][1]*100 ;
+      if (!(keys.includes(row.interaction.turn[key]))){ 
```

Comment:
It might help clarify here to use a variable for `row.interaction.turn[key]`so it doesn't have to repeat a few times on the same lines.  Maybe `rowKey`?